include/unifex/linux/io_uring_context.hpp

```diff
@@ -942,6 +956,171 @@ inline io_uring_context::scheduler io_uring_context::get_scheduler() noexcept {
+
+      auto populateSqe = [this](io_uring_sqe& sqe) noexcept {
+        sqe.opcode = IORING_OP_ACCEPT;
+        sqe.accept_flags = SOCK_NONBLOCK;
+        sqe.fd = fd_;
+
+        sqe.user_data = reinterpret_cast<std::uintptr_t>(
+            static_cast<completion_base*>(this));
+
+        this->execute_ = &operation::on_accept;
+      };
+
+      if (!context_.try_submit_io(populateSqe)) {
+        this->execute_ = &operation::on_schedule_complete;
+        context_.schedule_pending_io(this);
+      }
+    }
+
+    static void on_accept(operation_base* op) noexcept {
+      auto& self = *static_cast<operation*>(op);
+      if (self.result_ >= 0) {
+        if constexpr (noexcept(unifex::set_value(
+                          std::move(self.receiver_), async_read_write_file{self.context_, self.result_}))) {
+          unifex::set_value(std::move(self.receiver_), async_read_write_file{self.context_, self.result_});
+        } else {
+          UNIFEX_TRY {
+            unifex::set_value(std::move(self.receiver_), async_read_write_file{self.context_, self.result_});
+          }
+          UNIFEX_CATCH(...) {
+            unifex::set_error(
+                std::move(self.receiver_), std::current_exception());
+          }
+        }
+      } else if (self.result_ == -ECANCELED) {
+        unifex::set_done(std::move(self.receiver_));
+      } else {
+        unifex::set_error(
+            std::move(self.receiver_),
+            std::error_code{-self.result_, std::system_category()});
+      }
+    }
+
+    io_uring_context& context_;
+    int fd_;
+    Receiver receiver_;
+  };
+
+public:
+  // Produces number of bytes read.
+  template <
+      template <typename...>
+      class Variant,
+      template <typename...>
+      class Tuple>
+  using value_types = Variant<Tuple<ssize_t>>;
+
+  // Note: Only case it might complete with exception_ptr is if the
+  // receiver's set_value() exits with an exception.
+  template <template <typename...> class Variant>
+  using error_types = Variant<std::error_code, std::exception_ptr>;
+
+  static constexpr bool sends_done = true;
+
+  explicit accept_sender(io_uring_context& context, int fd) noexcept
+    : context_(context)
+    , fd_(fd) {}
+
+  template <typename Receiver>
+  operation<remove_cvref_t<Receiver>> connect(Receiver&& r) && {
+    return operation<remove_cvref_t<Receiver>>{*this, (Receiver &&) r};
+  }
+
+private:
+  io_uring_context& context_;
+  int fd_;
+};
+
+class io_uring_context::accept_stream {
+public:
+  using offset_t = std::int64_t;
+
+  explicit accept_stream(io_uring_context& context, std::int32_t port) noexcept
+    : context_(context)
+    , port_(port) {}
+
+  auto next() noexcept {
+    return let_value_with(
+        [this]() noexcept {
+          if (!fd_.valid()) {
+            open_socket();
+          }
+          return fd_.get();
+        },
+        [this](auto fd) noexcept { return accept_sender{context_, fd}; });
```

Comment:
Would this be cleaner if you did the "if not valid, open a socket" dance inside the `accept_sender` constructor?

Reply:
added a TODO